src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -902,3 +915,31 @@ def _rehome(self, l):
       return HermeticDistribution('.jdk', local_distribution)
     else:
       return local_distribution
+
+  def  _collect_invalid_compile_dependencies(self, compile_target, invalid_target_set):
```

Comment:
This method is fairly large, which makes me even more hesitant than usual to agree that overriding a concrete method is a good idea. Could you extract a smaller predicate or something?